src/Microsoft.VisualStudio.ProjectSystem.Managed/MsBuildUtilities.cs

```diff
@@ -0,0 +1,172 @@
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        /// <returns>Collection of individual values in the property.</returns>
+        public static ImmutableArray<string> GetPropertyValues(string propertyValue, char delimiter = ';')
+        {
+            var values = propertyValue.Split(delimiter).Select(f => f.Trim());
+
+            // We need to ensure that we return values in the specified order.
+            var valuesBuilder = ImmutableArray.CreateBuilder<string>();
+            foreach (var value in values)
+            {
+                if (!string.IsNullOrEmpty(value))
+                {
+                    valuesBuilder.Add(value);
+                }
+            }
+
+            return valuesBuilder.Distinct(StringComparer.OrdinalIgnoreCase).ToImmutableArray();
+        }
+
+        /// <summary>
+        /// Appends a value to a delimited property. If the property does not exist it will be added.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="evaluatedPropertyValue">Original evaluated value of the property.</param>
+        /// <param name="propertyName">Property name.</param>
+        /// <param name="propertyValue">Value to add to the property.</param>
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        public static void AppendPropertyValue(ProjectRootElement project, string evaluatedPropertyValue, string propertyName, string propertyValue, char delimiter = ';')
+        {
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(evaluatedPropertyValue, nameof(evaluatedPropertyValue));
+            Requires.NotNullOrEmpty(propertyName, nameof(propertyName));
+
+            var property = GetOrAddProperty(project, propertyName);
+            StringBuilder newValue = new StringBuilder();
+            foreach (string value in GetPropertyValues(evaluatedPropertyValue, delimiter))
+            {
+                newValue.Append(value);
+                newValue.Append(delimiter);
+            }
+
+            newValue.Append(propertyValue);
+            property.Value = newValue.ToString();
+        }
+
+        /// <summary>
+        /// Renames a value inside a delimited property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="evaluatedPropertyValue">Original evaluated value of the property.</param>
+        /// <param name="propertyName">Property name.</param>
+        /// <param name="propertyValue">Value to remove from the property.</param>
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        /// <remarks>
+        /// If the property is not present it will be added. This means that the evaluated property
```

Comment:
Look at my explanation on the other method. Same as there I don't feel strongly about it either way so I'm open to changing it to whatever makes the most sense.